graphene_subscriptions/consumers.py

```diff
@@ -45,14 +38,39 @@ def disconnect(self, close_code):
             )
 
     def receive_json(self, request):
+
         id = request.get("id")
 
         if request["type"] == "connection_init":
+            headers = dict(request["payload"])
+            if "Authorization" in headers:
+                    token_name, token_key = headers["Authorization"].split()
+                    if token_name == "JWT":
+                        token = jwt_decode(
+                            token_key,
+                            settings.SECRET_KEY,
+                            algorithms=["HS256"]
+                        )
+                        self.user = get_user_model().objects.get(
+                            username=token["username"]
+                        )
+                        close_old_connections()
             return
 
         elif request["type"] == "start":
+            
             payload = request["payload"]
-            context = AttrDict(self.scope)
+
+            temp = {}
+            for x, y in self.scope["headers"]:
+                temp[x.decode()]=y
+            self.scope["headers"] = temp
+
+            self.scope["method"] = "WebSocket"
+
+            context = AsgiRequest(self.scope, BytesIO(b""))
+
+            context.user = self.user 
```

Comment:
In case no header was provided - this line will result in a crash since there is no such class variable, am I right?
More cases should be tested and handled I believe

Reply:
I agree.
It's more of a sketch of a solution inspired by experiments with the library.

It also seems to me that I was having problems until I put the dictionary under init.

def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs).
        self.groups = {}